crypto/CMakeLists.txt

```diff
@@ -525,7 +525,21 @@ if(FIPS_SHARED)
             WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
             DEPENDS fips_empty_main ${PROJECT_SOURCE_DIR}/util/fipstools/capture_hash/capture_hash.go
             )
-    add_library(generated_fipsmodule OBJECT generated_fips_shared_support.c)
+    add_library(
+      generated_fipsmodule
+
+      OBJECT
+
+      generated_fips_shared_support.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpucap.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_arm_linux.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_aarch64_win.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_aarch64_apple.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_aarch64_linux.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_aarch64_fuchsia.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_intel.c
+      ${CMAKE_CURRENT_SOURCE_DIR}/fipsmodule/cpucap/cpu_ppc64le.c
```

Comment:
the Windows FIPS build is different than Linux and macOS, for Windows the module is compiled twice: first we compile the module, then we run a dummy test program which prints out the integrity hash value, then we compile again the module but this time with the integrity hash inserted into a helper file. Since cpucap files are not in `bcm.c` anymore we need to explicitly add them to the build files for the module, both in `fipsmodule/CMakeLists.txt` where the first compilation happens and in this CMake file where the second compilation happens.